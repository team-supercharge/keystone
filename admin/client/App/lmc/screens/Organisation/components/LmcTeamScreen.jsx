import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

export class LmcTeamScreen extends Component {
    render() {
        return (
            <div>
                Team
            </div>
        )
    }
}

LmcTeamScreen.propTypes = {}

const mapStateToProps = state => {
    return {}
}

const mapDispatchToProps = dispatch => {
    return {}
}

export default connect(mapStateToProps, mapDispatchToProps)(LmcTeamScreen)