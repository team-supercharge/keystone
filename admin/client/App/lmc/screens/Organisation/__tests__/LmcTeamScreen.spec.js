jest.mock('../../../../shared/CreateForm')

import React from 'react'
import { shallow } from 'enzyme'
import { LmcTeamScreen } from '../components/LmcTeamScreen.jsx'

describe('LmcTeamScreen', () => {
    let wrapper
    let users
    let selectedUser
    const setSelectedUser = jest.fn()
    const fetchUsers = jest.fn()

    beforeEach(() => {
        users = [
            { name: { first: 'Test', last: 'Carer' }, active: true },
            { name: { first: 'Test2', last: 'Carer2' }, active: true }
        ]
        selectedUser = 'TestId'

        wrapper = shallow(
            <LmcTeamScreen
                fetchUsers={fetchUsers}
                users={users}
                selectedUser={selectedUser}
                setSelectedUser={setSelectedUser}
            />
        )
    })

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('renders an LmcSidebar with the correct props', () => {
        const sidebar = wrapper.find('LmcSidebar')
        expect(sidebar.props().items).toEqual(users)
        expect(sidebar.props().selectedItem).toEqual(selectedUser)
        expect(sidebar.props().setSelectedItem).toEqual(setSelectedUser)
    })

    it('renders a spinner if no data is loaded', () => {
        const emptyWrapper = shallow(
            <LmcTeamScreen 
                fetchUsers={fetchUsers}
                users={null}
                selectedUser={selectedUser}
                setSelectedUser={setSelectedUser}
            />
        )
        expect(wrapper.find('LmcSpinner').length).toEqual(0)
        expect(emptyWrapper.find('LmcSpinner').length).toEqual(1)
    })

    it('renders a message if there are no team members yet, else renders profile', () => {
        const emptyWrapper = shallow(
            <LmcTeamScreen
                fetchUsers={fetchUsers}
                users={[]}
                selectedUser={null}
                setSelectedUser={setSelectedUser}
            />
        )

        expect(wrapper.find('BlankState').length).toEqual(0)
        expect(wrapper.find('Connect(LmcTeamMemberProfile)').length).toEqual(1)

        const message = emptyWrapper.find('BlankState')
        expect(message.props().heading).toEqual("You haven't added any team members yet")
        expect(emptyWrapper.find('LmcTeamMemberProfile').length).toEqual(0)
    })
})