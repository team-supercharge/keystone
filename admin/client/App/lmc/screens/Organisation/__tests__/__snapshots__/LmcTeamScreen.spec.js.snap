// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LmcTeamScreen renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <div
      style={
        Object {
          "flex": "1",
        }
      }
    >
      <LmcSidebar
        itemLabel="Team Member"
        items={
          Array [
            Object {
              "active": true,
              "name": Object {
                "first": "Test",
                "last": "Carer",
              },
            },
            Object {
              "active": true,
              "name": Object {
                "first": "Test2",
                "last": "Carer2",
              },
            },
          ]
        }
        listId="User"
        onCreate={[Function]}
        selectedItem="TestId"
        setSelectedItem={[MockFunction]}
        styles={
          Object {
            "height": "83vh",
            "maxHeight": "83vh",
          }
        }
        title="Team Members"
      />
    </div>
    <div
      style={
        Object {
          "flex": "3.5",
          "height": "83vh",
          "overflow": "scroll",
          "padding": "0px 20px 0px 20px",
          "wordWrap": "break-word",
        }
      }
    >
      <div
        style={
          Object {
            "margin": "0 auto",
            "maxWidth": 1000,
            "paddingTop": 50,
          }
        }
      >
        <Connect(LmcTeamMemberProfile)
          selectedUser="TestId"
        />
      </div>
    </div>
  </div>,
  "nodes": Array [
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <div
        style={
          Object {
            "flex": "1",
          }
        }
      >
        <LmcSidebar
          itemLabel="Team Member"
          items={
            Array [
              Object {
                "active": true,
                "name": Object {
                  "first": "Test",
                  "last": "Carer",
                },
              },
              Object {
                "active": true,
                "name": Object {
                  "first": "Test2",
                  "last": "Carer2",
                },
              },
            ]
          }
          listId="User"
          onCreate={[Function]}
          selectedItem="TestId"
          setSelectedItem={[MockFunction]}
          styles={
            Object {
              "height": "83vh",
              "maxHeight": "83vh",
            }
          }
          title="Team Members"
        />
      </div>
      <div
        style={
          Object {
            "flex": "3.5",
            "height": "83vh",
            "overflow": "scroll",
            "padding": "0px 20px 0px 20px",
            "wordWrap": "break-word",
          }
        }
      >
        <div
          style={
            Object {
              "margin": "0 auto",
              "maxWidth": 1000,
              "paddingTop": 50,
            }
          }
        >
          <Connect(LmcTeamMemberProfile)
            selectedUser="TestId"
          />
        </div>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <LmcTeamScreen
        fetchUsers={[MockFunction]}
        selectedUser="TestId"
        setSelectedUser={[MockFunction]}
        users={
          Array [
            Object {
              "active": true,
              "name": Object {
                "first": "Test",
                "last": "Carer",
              },
            },
            Object {
              "active": true,
              "name": Object {
                "first": "Test2",
                "last": "Carer2",
              },
            },
          ]
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": LmcTeamScreen {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "onCreateUserComplete": [Function],
        "props": Object {
          "fetchUsers": [MockFunction],
          "selectedUser": "TestId",
          "setSelectedUser": [MockFunction],
          "users": Array [
            Object {
              "active": true,
              "name": Object {
                "first": "Test",
                "last": "Carer",
              },
            },
            Object {
              "active": true,
              "name": Object {
                "first": "Test2",
                "last": "Carer2",
              },
            },
          ],
        },
        "refs": Object {},
        "renderDesktop": [Function],
        "renderMobile": [Function],
        "renderUserInfo": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <div
            style={
              Object {
                "flex": "1",
              }
            }
          >
            <LmcSidebar
              itemLabel="Team Member"
              items={
                Array [
                  Object {
                    "active": true,
                    "name": Object {
                      "first": "Test",
                      "last": "Carer",
                    },
                  },
                  Object {
                    "active": true,
                    "name": Object {
                      "first": "Test2",
                      "last": "Carer2",
                    },
                  },
                ]
              }
              listId="User"
              onCreate={[Function]}
              selectedItem="TestId"
              setSelectedItem={[MockFunction]}
              styles={
                Object {
                  "height": "83vh",
                  "maxHeight": "83vh",
                }
              }
              title="Team Members"
            />
          </div>
          <div
            style={
              Object {
                "flex": "3.5",
                "height": "83vh",
                "overflow": "scroll",
                "padding": "0px 20px 0px 20px",
                "wordWrap": "break-word",
              }
            }
          >
            <div
              style={
                Object {
                  "margin": "0 auto",
                  "maxWidth": 1000,
                  "paddingTop": 50,
                }
              }
            >
              <Connect(LmcTeamMemberProfile)
                selectedUser="TestId"
              />
            </div>
          </div>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <div
            style={
              Object {
                "flex": "1",
              }
            }
          >
            <LmcSidebar
              itemLabel="Team Member"
              items={
                Array [
                  Object {
                    "active": true,
                    "name": Object {
                      "first": "Test",
                      "last": "Carer",
                    },
                  },
                  Object {
                    "active": true,
                    "name": Object {
                      "first": "Test2",
                      "last": "Carer2",
                    },
                  },
                ]
              }
              listId="User"
              onCreate={[Function]}
              selectedItem="TestId"
              setSelectedItem={[MockFunction]}
              styles={
                Object {
                  "height": "83vh",
                  "maxHeight": "83vh",
                }
              }
              title="Team Members"
            />
          </div>
          <div
            style={
              Object {
                "flex": "3.5",
                "height": "83vh",
                "overflow": "scroll",
                "padding": "0px 20px 0px 20px",
                "wordWrap": "break-word",
              }
            }
          >
            <div
              style={
                Object {
                  "margin": "0 auto",
                  "maxWidth": 1000,
                  "paddingTop": 50,
                }
              }
            >
              <Connect(LmcTeamMemberProfile)
                selectedUser="TestId"
              />
            </div>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LmcTeamScreen
    fetchUsers={[MockFunction]}
    selectedUser="TestId"
    setSelectedUser={[MockFunction]}
    users={
      Array [
        Object {
          "active": true,
          "name": Object {
            "first": "Test",
            "last": "Carer",
          },
        },
        Object {
          "active": true,
          "name": Object {
            "first": "Test2",
            "last": "Carer2",
          },
        },
      ]
    }
  />,
}
`;
